src/renderers/shared/stack/reconciler/ReactCompositeComponent.js

```diff
@@ -1100,10 +1100,17 @@ var ReactCompositeComponent = {
    * @final
    * @private
    */
-  attachRef: function(ref, component) {
+  attachRef: function(ref, component, mockConfig) {
     var inst = this.getPublicInstance();
     invariant(inst != null, 'Stateless function components cannot have refs.');
-    var publicComponentInstance = component.getPublicInstance();
+    var publicComponentInstance;
+    if (mockConfig && mockConfig.getMockRef) {
```

Comment:
`ReactCompositeComponent` should not be aware of test renderer-specific APIs. Why is defining `getPublicInstance()` inside test renderer not enough?